File: objectModel/CSharp/Microsoft.CommonDataModel.ObjectModel.Tests/AdlsTestHelper.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

using Microsoft.CommonDataModel.ObjectModel.Storage;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using System;

namespace Microsoft.CommonDataModel.ObjectModel.Tests
{
    /// <summary>
    /// Class to initialize adls tests adapters.
    /// </summary>
    public static class AdlsTestHelper
    {
        public static void CheckADLSEnvironment()
        {
            if (String.IsNullOrEmpty(Environment.GetEnvironmentVariable("ADLS_RUNTESTS")))
            {
                // this will cause tests to appear as "Skipped" in the final result
                Assert.Inconclusive("ADLS environment not set up");
            }
        }

        public static ADLSAdapter CreateAdapterWithSharedKey(string rootRelativePath = null, bool testBlobHostName = false)
        {
            string hostname = Environment.GetEnvironmentVariable("ADLS_HOSTNAME");
            string rootPath = Environment.GetEnvironmentVariable("ADLS_ROOTPATH");
            string sharedKey = Environment.GetEnvironmentVariable("ADLS_SHAREDKEY");

            Assert.IsFalse(String.IsNullOrEmpty(hostname), "ADLS_ENDPOINT not set");
            Assert.IsFalse(String.IsNullOrEmpty(rootPath), "ADLS_ROOTPATH not set");
            Assert.IsFalse(String.IsNullOrEmpty(sharedKey), "ADLS_SHAREDKEY not set");

            if (testBlobHostName)
            {
                hostname = hostname.Replace("dfs", "blob");
            }

            return new ADLSAdapter(hostname, GetFullRootPath(rootPath, rootRelativePath), sharedKey);
        }

        public static ADLSAdapter CreateAdapterWithClientId(string rootRelativePath = null, bool specifyEndpoint = false, bool testBlobHostName = false)
        {
            string hostname = Environment.GetEnvironmentVariable("ADLS_HOSTNAME");
            string rootPath = Environment.GetEnvironmentVariable("ADLS_ROOTPATH");
            string tenant = Environment.GetEnvironmentVariable("ADLS_TENANT");
            string clientId = Environment.GetEnvironmentVariable("ADLS_CLIENTID");
            string clientSecret = Environment.GetEnvironmentVariable("ADLS_CLIENTSECRET");

            Assert.IsFalse(String.IsNullOrEmpty(hostname), "ADLS_ENDPOINT not set");
            Assert.IsFalse(String.IsNullOrEmpty(rootPath), "ADLS_ROOTPATH not set");
            Assert.IsFalse(String.IsNullOrEmpty(tenant), "ADLS_TENANT not set");
            Assert.IsFalse(String.IsNullOrEmpty(clientId), "ADLS_CLIENTID not set");
            Assert.IsFalse(String.IsNullOrEmpty(clientSecret), "ADLS_CLIENTSECRET not set");

            if (testBlobHostName)
            {
                hostname = hostname.Replace("dfs", "blob");
            }

            if (specifyEndpoint)
            {
                return new ADLSAdapter(hostname, GetFullRootPath(rootPath, rootRelativePath), tenant, clientId, clientSecret, Enums.AzureCloudEndpoint.AzurePublic);
            }

            return new ADLSAdapter(hostname, GetFullRootPath(rootPath, rootRelativePath), tenant, clientId, clientSecret);
        }

        public static string GetFullRootPath(string rootPath, string rootRelativePath)
        {
            if (rootRelativePath == null)
            {
                return rootPath;
            }

            if (rootPath.EndsWith("/"))
            {
                rootPath = rootPath.Substring(0, rootPath.Length - 1);
            }

            if (rootRelativePath.StartsWith("/"))
            {
                rootRelativePath = rootRelativePath.Substring(1);
            }

            return $"{rootPath}/{rootRelativePath}";
        }
    }
}

File: objectModel/Java/objectmodel/src/test/java/com/microsoft/commondatamodel/objectmodel/AdlsTestHelper.java
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License. See License.txt in the project root for license information.

package com.microsoft.commondatamodel.objectmodel;

import com.microsoft.commondatamodel.objectmodel.enums.AzureCloudEndpoint;
import com.microsoft.commondatamodel.objectmodel.storage.AdlsAdapter;
import com.microsoft.commondatamodel.objectmodel.utilities.StringUtils;
import org.testng.SkipException;

import static org.testng.Assert.assertFalse;

public class AdlsTestHelper {

    public static void checkADLSEnvironment()
    {
        if (!"1".equals(System.getenv("ADLS_RUNTESTS")))
        {
            // this will cause tests to appear as "Skipped" in the final result
            throw new SkipException("ADLS environment not set up");
        }
    }

    public static AdlsAdapter createAdapterWithSharedKey() {
        return createAdapterWithSharedKey("", false);
    }

    public static AdlsAdapter createAdapterWithSharedKey(String rootRelativePath, boolean testBlobHostname) {
        String hostname = System.getenv("ADLS_HOSTNAME");
        String rootPath = System.getenv("ADLS_ROOTPATH");
        String sharedKey = System.getenv("ADLS_SHAREDKEY");

        assertFalse(StringUtils.isNullOrEmpty(hostname));
        assertFalse(StringUtils.isNullOrEmpty(rootPath));
        assertFalse(StringUtils.isNullOrEmpty(sharedKey));

        if (testBlobHostname) {
            hostname = hostname.replace("dfs", "blob");
        }

        return new AdlsAdapter(hostname, getFullRootPath(rootPath, rootRelativePath), sharedKey);
    }

    public static AdlsAdapter createAdapterWithClientId() {
        return createAdapterWithClientId("", false, false);
    }

    public static AdlsAdapter createAdapterWithClientId(boolean specifyEndpoint) {
        return createAdapterWithClientId("", specifyEndpoint, false);
    }

    public static AdlsAdapter createAdapterWithClientId(String rootRelativePath) {
        return createAdapterWithClientId(rootRelativePath, false, false);
    }

    public static AdlsAdapter createAdapterWithClientId(String rootRelativePath, boolean specifyEndpoint, boolean testBlobHostname) {
        String hostname = System.getenv("ADLS_HOSTNAME");
        String rootPath = System.getenv("ADLS_ROOTPATH");
        String tenant = System.getenv("ADLS_TENANT");
        String clientId = System.getenv("ADLS_CLIENTID");
        String clientSecret = System.getenv("ADLS_CLIENTSECRET");

        assertFalse(StringUtils.isNullOrEmpty(hostname));
        assertFalse(StringUtils.isNullOrEmpty(rootPath));
        assertFalse(StringUtils.isNullOrEmpty(tenant));
        assertFalse(StringUtils.isNullOrEmpty(clientId));
        assertFalse(StringUtils.isNullOrEmpty(clientSecret));

        if (testBlobHostname) {
            hostname = hostname.replace("dfs", "blob");
        }

        if (specifyEndpoint) {
            return new AdlsAdapter(hostname, getFullRootPath(rootPath, rootRelativePath), tenant, clientId, clientSecret, AzureCloudEndpoint.AzurePublic);
        }

        return new AdlsAdapter(hostname, getFullRootPath(rootPath, rootRelativePath), tenant, clientId, clientSecret);
    }

    public static String getFullRootPath(String rootPath, String rootRelativePath)
    {
        if (rootRelativePath == null || rootRelativePath.isEmpty()) {
            return rootPath;
        }

        if (rootPath.endsWith("/")) {
            rootPath = rootPath.substring(0, rootPath.length() - 1);
        }

        if (rootRelativePath.startsWith("/")) {
            rootRelativePath = rootRelativePath.substring(1);
        }

        return rootPath + "/" + rootRelativePath;
    }
}
